### ISSUE TITLE
False negative for `UseIsNullOrEmpty` rule

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6024]
For the below test cases

```kotlin
@Test
fun `chained call`() {
    val code = """
        class A {
            val t: String? = ""
        }
        fun test() {
            val a = A()
            if (a.t == null || 0 == a.t.length) return
        }
    """.trimIndent()
    val findings = subject.compileAndLintWithContext(env, code)
    assertThat(findings).hasSize(1)
}
```

```kotlin
@Test
fun `chained call with package string`() {
    java.io.File.separator
    val code = """
        fun test() {
            if (java.io.File.separator == null || java.io.File.separator == 0) return
        }
    """.trimIndent()
    val findings = subject.compileAndLintWithContext(env, code)
    assertThat(findings).hasSize(1)
}
```

## Expected Behavior
TC should pass

## Observed Behavior
TC is failing

## Context
It seems in case when variable a qualified expression before it's use(either a package or class) then rule is failing
